ComputeDataDrivenCrossSection.py

```diff
@@ -158,11 +183,11 @@
 
     if args.FD:
         crossSec, crossSecUnc = ComputeCrossSection(rawYield, rawYieldUnc, frac, uncFrac, effAcc,
-                                                    ptMax - ptMin, 1., sigmaMB, nEv, 1., 'corr')
+                                                    ptMax - ptMin, 1., sigmaMB, nEv, 1., RYFractionCorrelation)
     else:
         # TODO: check if uncorrelated is the right option or anti-correlated is better
         crossSec, crossSecUnc = ComputeCrossSection(rawYield, rawYieldUnc, frac, uncFrac, effAcc,
-                                                    ptMax - ptMin, 1., sigmaMB, nEv, 1., 'uncorr')
+                                                    ptMax - ptMin, 1., sigmaMB, nEv, 1., RYFractionCorrelation) # TODO: check this
```

Comment:
Here the if-else is not needed since you are doing the same thing for prompt and FD. Personally, I would set up the macro to have as default `corr` for feed-down and `uncorr` for prompt, then you can change the propagation assumption from the command line.